docs/howto/custom-shell.txt

```diff
@@ -0,0 +1,46 @@
+====================================== 
+How to customize the ``shell`` command 
+======================================
+
+To customize the :djadmin:`shell`, create a new :doc:`management command
+</howto/custom-management-commands>` which subclasses 
+``django.core.management.commands.shell.Command`` and overrides the existing
+``shell`` management command. See :ref:`Overriding commands 
+<overriding-commands>` for more details. 
+
+Customizing the auto-imports
+============================
+
+.. versionadded:: 5.2
+
+To customize the auto-importing behavior of :djadmin:`shell`, you can override
+the ``get_namespace()`` method. For example, assuming you have a ``polls``
+application in :setting:`INSTALLED_APPS`:
```

Comment:
Now thinking the "assuming you have a ``polls`` application in :setting:`INSTALLED_APPS`" is distracting

```suggestion
To customize the auto-importing behavior of the :djadmin:`shell` management
command, you can override the ``get_namespace()`` method.

For example:
```